### ISSUE TITLE
Can't fork a local node from avax public rpc

### ISSUE THREAD

**Issue by Ajezezzzz** [NomicFoundation/hardhat#5080]
### Version of Hardhat

2.22.2

### What happened?

When I try to start a forked local node from the Avax network using the public RPC, I always receive the same error.
Error HH604: Error running JSON-RPC server: The Http server returned error status code: HTTP status server error (500 Internal Server Error) for url (https://api.avax-test.network/ext/bc/C/rpc)

### Minimal reproduction steps

My hardhat.config.js:
```
require("@nomicfoundation/hardhat-toolbox");

/** @type import('hardhat/config').HardhatUserConfig */
module.exports = {
  defaultNetwork: "hardhat",
  solidity: "0.8.24",
  networks: {
    hardhat: {
      forking: {
        url: "https://api.avax.network/ext/bc/C/rpc",
      },
    },
  }
};
```
Starting the node with the command:
`npx hardhat node`

### Search terms

avax avalanche hh604

**Comment by kanej** [comment]
@Ajezezzzz have you been able to fork against Avax in previous versions of Hardhat or is this the first time you are encountering this issue?

**Comment by Ajezezzzz** [comment]
Yes, I have other environment that work. But I need to keep my packagelock.json else, it doesn't work.

Here is the package.json:
```
{
  "name": "hardhat-project",
  "devDependencies": {
    "@nomicfoundation/hardhat-toolbox": "^4.0.0",
    "hardhat": "^2.19.4"
  },
  "dependencies": {
    "@nomicfoundation/hardhat-viem": "^2.0.0",
    "viem": "^2.7.13"
  }
}
```

And the package-lock.json:
[package-lock.json](https://github.com/NomicFoundation/hardhat/files/14875010/package-lock.json)

**Comment by Ajezezzzz** [comment]
Yes, I have other environment that work. But I need to keep my packagelock.json else, it doesn't work.

Here is the package.json:
```
{
  "name": "hardhat-project",
  "devDependencies": {
    "@nomicfoundation/hardhat-toolbox": "^4.0.0",
    "hardhat": "^2.19.4"
  },
  "dependencies": {
    "@nomicfoundation/hardhat-viem": "^2.0.0",
    "viem": "^2.7.13"
  }
}
```

And the package-lock.json:
[package-lock.json](https://github.com/NomicFoundation/hardhat/files/14875029/package-lock.json)
